Removes stale code from /hosting-config for starting the Creator trial

p-jackson opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/6638

## Proposed Changes

This is a sub part of https://github.com/Automattic/dotcom-forge/issues/6638, a task to remove code related to the old endpoint for starting trials. Doing multiple PRs otherwise the diff is quite large.

There used to be a trial button in `/hosting-config` which was originally disabled in #88486. But it left some of the old code in place. We need to remove it so we can eventually remove the `useAddHostingTrialMutation` hook and API endpoint.

~The hosting config page also uses the `WithOnclickTrialRequest` HoC, which from the name appears to have something to do with a trial, but when I looked at the code it's unrelated to the hosting trial.~
I've removed `WithOnclickTrialRequest`.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Review the code and test the hosting config page.
* There should be no changes to the hosting config page.
* Use `/hosting-config` to transfer a simple site to atomic.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

p-jackson committed:
Removes stale code from /hosting-config for starting the Creator trial

The trial button was originally removed in #88486, but we're removing
more code so we can eventually remove the useAddHostingTrialMutation
hook

vykes-mac reviewed (commented):
`WithOnclickTrialRequest` HOC can be removed also since it's no longer needed in hosting-config

chad1008 reviewed (commented):
These changes look good to me! Sorry for the delayed review :rocket: :ship:

p-jackson committed:
Remove WithOnclickTrialRequest

The page updates correctly when atomic transfer is complete, so it
doesn't need the logic from WithOnclickTrialRequest.

This HOC was added in #86588 to deal with automatically starting AT
after switching from Free to a Creator Trial.

p-jackson commented:
@vykes-mac I'm not really familiar with `WithOnclickTrialRequest` but it seems like it's there for refetching site data once atomic transfer is complete? It was added in #86588 though, so definitely trial related.

I removed the HOC and the regular method of atomic transfer still works as expected. I almost removed the entire `requestUpdatedSiteData` callback since it seemed like the UI should update itself when the transfer is complete. However the `<HostingActivateStatus>` banner momentarily showed the wrong state (the "Activate" button reappears) when transfer completes if I remove it.

https://github.com/Automattic/wp-calypso/assets/1500769/de53f19f-90c9-4c8b-993b-d13b7841eaf4

So I've left it there and just removed the code that refreshes site data when the transfer completes.

paulopmt1 commented:
Do we still want to pursue this @p-jackson?

p-jackson commented:
@paulopmt1 wow blast from the pre-sabbatical past! I do think we need to wrap this and https://github.com/Automattic/wp-calypso/pull/90152 up (and all of https://github.com/Automattic/dotcom-forge/issues/6638). I think leaving the rest API for trials exposed when we're not going to use it is a bad idea.

paulopmt1 commented:
> I'm not sure if I can count it towards Groundskeeping since it's not really user facing.

How do you see it @autumnfjeld?

autumnfjeld commented:
> I'm not sure if I can count it towards Groundskeeping since it's not really user facing.

I don't see any mention of work needing to be user facing in pdqkMK-1mK-p2

IMO, addressing tech debt is an important part of GK maintenance.

paulopmt1 commented:
> IMO, addressing tech debt is an important part of GK maintenance.

Nice, in that case, @p-jackson, could you rebase it when working on GK maintenance so I can review it?

p-jackson commented:
After attempting to rebase I realise the code I was trying to clean up was already removed during the untangling project. So my sorry, it looks like this PR _isn't_ relevant anymore 😅 Still a worthwhile conversation though to discuss what sort of things should count as GK.

https://github.com/Automattic/dotcom-forge/issues/6638 is still relevant, it's just that this sub-task is already done :)
